### PR TITLE
#193: Replace Jest with Vitest

### LINKED ISSUE #193: SPIKE: Replace Jest with Vitest
Investigate switching our testing framework from Jest to Vitest due to #187 and various other integration problems we've been facing between Jest and Vite throughout the project.

### PR DESCRIPTION
**Pull request by Subzeero**
Closes #193 

# What was the issue
We have been facing numerous compatibility issues between Vite and Jest throughout the development of this project. Vitest aims to fix many of these issues by integrating with Vite, but also retaining a Jest-compatible API (for the most part). 

# What was done and why
Removed Jest- and Babel-related packages and configuration. Replaced them with Vitest and necessary configuration.

# How it was tested
Patched our existing unit test cases to work with Vitest so they all pass.

### CODE REVIEW

**Comment by Subzeero** [comment]
It appears that after the merge, the test hook integration needs some adjusting. Will look into this.

**Comment by LongQuanNguyen** [comment]
On the smoke test spike, the test environment was set to node and all the jest test was running fine.

On this, the environment was set to jsdom, that is the reason writing to a file in the test with vitest can not run. However, when I set the environment to node, more tests failed...

We need higher power...

**Comment by LongQuanNguyen** [comment]
or a server

or separate the test for smoke tests to a folder outside of the app folder, and have the CICD pipeline run with a separate environment

File: my-webxr-app/src/testing/TextInFileSearch.tsx [hunk 1]
```diff
@@ -1,5 +1,11 @@
-import * as fs from 'fs';
-import * as readline from 'readline';
+// import * as fs from 'fs';
+// import * as readline from 'readline';
+
+// TODO: I don't know why this works
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fs = require('fs');
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const readline = require('readline');
```
Comment (LongQuanNguyen) [inline my-webxr-app/src/testing/TextInFileSearch.tsx:8]: I don't either, but it is what it is

Comment (LongQuanNguyen) [reply my-webxr-app/src/testing/TextInFileSearch.tsx:8]: @Subzeero merge quick before it breaks again

**Review by trangnguyen3010 — APPROVED** [review]

**Pull request merged**